sources/openmetrics/openmetrics.go

```diff
@@ -2,35 +2,156 @@ package openmetrics
+	IgnoredMetrics     util.Regexp   `yaml:"ignored_metrics"`
+	Interval           time.Duration `yaml:"interval"`
+	SummaryQuantileTag string        `yaml:"summary_quantile_tag"`
+}
+
+type OpenMetricsSource struct {
+	cancelFunc         context.CancelFunc
+	context            context.Context
+	histogramBucketTag string
+	Host               *url.URL
+	HttpClient         *http.Client
+	IgnoredMetrics     *regexp.Regexp
+	interval           time.Duration
+	logger             *logrus.Entry
+	name               string
+	server             *veneur.Server
+	summaryQuantileTag string
+}
+
+type QueryResults struct {
 	MetricFamily dto.MetricFamily
 	Error        error
 }
 
-func Query(
-	ctx context.Context, httpClient *http.Client, host string,
-	ignoredMetrics *regexp.Regexp,
-) (<-chan PrometheusResults, error) {
-	request, err := http.NewRequestWithContext(ctx, "GET", host, nil)
+type convertResults struct {
+	Metric *samplers.UDPMetric
+	Error  error
+}
+
+func ParseConfig(
+	name string, config interface{},
+) (veneur.ParsedSourceConfig, error) {
+	openMetricsSourceConfig := OpenMetricsSourceConfig{}
+	err := util.DecodeConfig(name, config, &openMetricsSourceConfig)
 	if err != nil {
 		return nil, err
 	}
-	response, err := httpClient.Do(request)
+
+	if openMetricsSourceConfig.HistogramBucketTag == "" {
+		openMetricsSourceConfig.HistogramBucketTag = "le"
+	}
+	if openMetricsSourceConfig.SummaryQuantileTag == "" {
+		openMetricsSourceConfig.SummaryQuantileTag = "quantile"
+	}
+
+	return openMetricsSourceConfig, nil
+}
+
+func Create(
+	server *veneur.Server, name string, logger *logrus.Entry,
+	sourceConfig veneur.ParsedSourceConfig,
+) (sources.Source, error) {
+	openMetricsSourceConfig, ok := sourceConfig.(OpenMetricsSourceConfig)
+	if !ok {
+		return nil, errors.New("invalid sink config type")
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	return OpenMetricsSource{
+		cancelFunc:         cancelFunc,
+		context:            ctx,
+		histogramBucketTag: openMetricsSourceConfig.HistogramBucketTag,
+		Host:               openMetricsSourceConfig.Host.Value,
+		HttpClient:         server.HTTPClient,
+		IgnoredMetrics:     openMetricsSourceConfig.IgnoredMetrics.Value,
+		interval:           openMetricsSourceConfig.Interval,
+		logger:             logger,
+		name:               name,
+		server:             server,
+		summaryQuantileTag: openMetricsSourceConfig.SummaryQuantileTag,
+	}, nil
+}
+
+func (source OpenMetricsSource) Name() string {
+	return source.name
+}
+
+func (source OpenMetricsSource) Start() error {
+	ticker := time.NewTicker(source.interval)
+intervalLoop:
+	for {
+		select {
+		case t := <-ticker.C:
+			ctx, cancel :=
+				context.WithDeadline(source.context, t.Add(source.interval))
+			results, err := source.Query(ctx)
+			cancel()
+			if err != nil {
+				source.logger.WithError(err).Warn("failed to query metrics")
+				continue
+			}
+			udpMetrics := source.Convert(results)
```

Comment:
A little weird we are calling this udpMetrics - but this may be the simplest path forward without a big refactor of Veneur.

Reply:
Yes, unfortunately this is the simplest path forward.